internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -308,10 +320,11 @@ func (r *AgentServiceConfigReconciler) Reconcile(origCtx context.Context, req ct
 		}
 	}
 
-	// Ensure image-service StatefulSet is reconciled
-	if err := ensureImageServiceStatefulSet(ctx, log, asc); err != nil {
-		return ctrl.Result{Requeue: true}, err
-
+	// Ensure image-service StatefulSet is reconciled (only if image service is not disabled)
+	if !isImageServiceDisabled(asc.Object.GetAnnotations()) {
```

Comment:
We should probably specifically remove the stateful set if the image service becomes disabled, right?
This looks like it only handles not creating it.

Reply:
I've made this annotation "immutable" so we don't need to worry about these cases. I think this normally is something you want from the start and never change, but if you have some use case in mind I can change that

Reply:
we've introduced the logic not long ago with the PVC/secret prefix feature: https://github.com/openshift/assisted-service/pull/7955/files#diff-c619bb36181b5e1dd76fb1fcc1f54068b16a8761f77106052fcbf5e2f5f9d356R2982
Basically it doesn't allow this annotation to be added after creation, nor modified. It must be either present or absent at creation and the value can never change. I think it fits well with this feature, would you agree?